common/util.go

```diff
@@ -931,8 +932,12 @@ func SleepWithMinDuration(desired time.Duration, available time.Duration) time.D
 	return available - d
 }
 
-func StringToRSAPublicKey(publicKeyString string) (*rsa.PublicKey, error) {
-	block, _ := pem.Decode([]byte(publicKeyString))
+func LoadRSAPublicKey(path string) (*rsa.PublicKey, error) {
+	key, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("invalid public key path %s", path)
+	}
+	block, _ := pem.Decode(key)
```

Comment:
What is the second return value? If it’s error then we should check and return if not nil

Reply:
https://pkg.go.dev/encoding/pem#Decode it returns the rest of the input after the key

An example here https://golang.org/src/encoding/pem/example_test.go  you can see it returns `"and some more"`  from line 31